tools/devtools/src/devtools/contexts/YorkieSource.tsx

```diff
@@ -121,12 +140,52 @@ export function useYorkieDoc() {
   return value;
 }
 
+export enum TransactionEventType {
+  Document = 'document',
+  Presence = 'presence',
+}
+
+export const getTransactionEventType = (
+  event: TransactionEvent,
+): TransactionEventType => {
+  for (const docEvent of event) {
+    if (
+      docEvent.type === DocEventType.StatusChanged ||
+      docEvent.type === DocEventType.Snapshot ||
+      docEvent.type === DocEventType.LocalChange ||
+      docEvent.type === DocEventType.RemoteChange
+    ) {
+      return TransactionEventType.Document;
+    }
+  }
+
+  return TransactionEventType.Presence;
+};
+
 export function useTransactionEvents() {
-  const value = useContext(TransactionEventsContext);
-  if (value === undefined) {
+  const { events, hidePresenceEvents, setHidePresenceEvents } = useContext(
+    TransactionEventsContext,
+  );
+
+  if (events === undefined) {
     throw new Error(
       'useTransactionEvents should be used within YorkieSourceProvider',
     );
   }
-  return value;
+
+  const filteredEvents = useMemo(
+    () =>
+      events.filter(
+        (event) =>
+          !hidePresenceEvents ||
+          getTransactionEventType(event) === TransactionEventType.Document,
+      ),
```

Comment:
How about modifying the event filtering to only apply when `hidePresenceEvents` is true? This would make the logic more explicit and potentially improve performance.

```js
if (!hidePresenceEvents) {
  return events;
}

return events.filter((event) => getTransactionEventType(event) === TransactionEventType.Document);
```

Reply:
Absolutely, this is much better option! Clear to read, and much better in performance! I'll make a change.